hamilton/execution/graph_functions.py

```diff
@@ -270,7 +260,13 @@ def execute_lifecycle_for_node(
                     raise
                 finally:
```

Comment:
I played around and got a deterministic outcome for this try-except-finally clause:
1) If the node executes normally, finally does not get called
2) If node raises exception finally gets called

Reply:
Opposed to this case where there is no ValueError.
```
2024-08-21 23:43:49,243 INFO worker.py:1781 -- Started a local Ray instance.

Capturing execution run. Results can be found at http://localhost:8241/dashboard/project/1/runs/65


Captured execution run. Results can be found at http://localhost:8241/dashboard/project/1/runs/65

(execute_lifecycle_for_node pid=361616) try lifecycle
(execute_lifecycle_for_node pid=361616) pre-hook
(execute_lifecycle_for_node pid=361616) try pre-hook
(execute_lifecycle_for_node pid=361616) node execute as callable
(execute_lifecycle_for_node pid=361616) 5s executed
    node_5s
0  5.025094
```